bindinfo/handle.go

```diff
@@ -128,24 +130,25 @@ func (h *BindHandle) Update(fullLoad bool) (err error) {
 	h.bindInfo.Lock()
 	newCache := h.bindInfo.Value.Load().(cache).copy()
 	defer func() {
+		h.bindInfo.lastUpdateTime = lastUpdateTime
 		h.bindInfo.Value.Store(newCache)
 		h.bindInfo.Unlock()
 	}()
 
 	for _, row := range rows {
 		hash, meta, err := h.newBindRecord(row)
 		// Update lastUpdateTime to the newest one.
-		if meta.Bindings[0].UpdateTime.Compare(h.lastUpdateTime) > 0 {
-			h.lastUpdateTime = meta.Bindings[0].UpdateTime
+		if meta.Bindings[0].UpdateTime.Compare(lastUpdateTime) > 0 {
+			lastUpdateTime = meta.Bindings[0].UpdateTime
 		}
 		if err != nil {
 			logutil.BgLogger().Error("update bindinfo failed", zap.Error(err))
 			continue
 		}
 
 		oldRecord := newCache.getBindRecord(hash, meta.OriginalSQL, meta.Db)
-		newRecord := merge(oldRecord, meta)
-		if meta.HasUsingBinding() {
+		newRecord := merge(oldRecord, meta).removeDeletedBindings()
+		if len(newRecord.Bindings) > 0 {
```

Comment:
Is this change a bug fix? Since we have `Bindings` which status are "pending verify".

Reply:
Yes.